### PR TITLE
Feature/contact form

### PR DESCRIPTION
**Pull request by raymondafan**
## Describe your changes
Created the contact form component which appears when a button is clicked. When a user clicks outside of the contact form, the modal closes. There is also validation for each input. A story for this component has been made as well.

## Issue ticket number and link
https://tripleten-apiary.atlassian.net/browse/EES-21 EES-21
## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: src/components/CheckboxInput/CheckboxInput.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  required?: boolean; // Optional prop for making the checkbox required
+  requiredError?: string; // Custom error message for the required validation
+  error?: { message: string };
+}
+
+const CheckboxInput: React.FC<CheckboxInputProps> = ({
+  name,
+  label,
+  required = false, // Default is not required
+  requiredError = "You must agree to the terms", // Default error message
```
Comment (XDRO) [inline src/components/CheckboxInput/CheckboxInput.tsx]: I believe this would look better if it were aligned with the rest of the check box input text. 
image

File: src/components/ContactForm/ContactForm.tsx
Comment (XDRO) [inline src/components/ContactForm/ContactForm.tsx:1]: This component lacks responsiveness at small dimensions starting at 479px. I've attached a recording.

https://github.com/user-attachments/assets/aa0d72ca-c454-49e1-8a89-132962377e18

File: src/components/ContactForm/ContactForm.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  title: "Components/ContactForm",
+  component: ContactForm,
+} as Meta;
+
+const Template: StoryFn<{ isOpen: boolean }> = ({ isOpen }) => {
+  const [modalOpen, setModalOpen] = useState(isOpen);
+
+  return (
+    <>
+      <button onClick={() => setModalOpen(true)}>Open Contact Form</button>
```
Comment (XDRO) [inline src/components/ContactForm/ContactForm.stories.tsx]: Not a necessary change, but one that I believe would enhance the storybooks user experience, would be to add some styling to this Open contact form button.
image 
Here is what it looks like from the Modal stories tsx file 
image
Which you can just take this code here to add some styling to this component:
        className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-full"
 This will make it match with the modal stories tsx button. (Attached in the 2nd screenshot)

File: src/components/ContactForm/ContactForm.stories.tsx
Comment (XDRO) [inline src/components/ContactForm/ContactForm.stories.tsx:1]: The controls do not work for this component either. You can take a look at my modal stories component at how to implement this functionality, with useArgs()

https://github.com/user-attachments/assets/942a490d-381a-4cec-8bed-12828b97e8ef

**Review by XDRO — COMMENTED** [review]
There are a few minor changes to be made, once addressed I will approve.

**Comment by raymondafan** [comment]
Appreciate the reviews @XDRO ! Just went through all of them. Let me know if there are any other problems.

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -8,13 +8,13 @@ export const Button = ({ onClick, buttonText, submitButton }: Props) => {
   return (
     <>
       <button
-        className={`group relative w-80 h-20 rounded-3xl border-6 border-lightGingerFlower bg-lightGingerFlower md:border-4 md:rounded-2xl md:w-56 md:h-14`}
+        className={` group relative w-80 h-20 rounded-3xl border-6 border-lightGingerFlower bg-lightGingerFlower md:border-4 md:rounded-2xl md:w-56 md:h-14 `}
```
Comment (tomershvueli) [inline src/components/Button/Button.tsx]: Let's undo the changes to this file since these whitespace changes are unnecessary and may cause merge issues down the road.

File: src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -42,16 +42,16 @@ export const Modal = ({ children, name, isOpen, onClose }: ModalProps) => {
 
   return (
     <ModalOverlay
-      className={`fixed inset-0 bg-black bg-opacity-50 flex justify-center items-center ${
+      className={`fixed inset-0 bg-black bg-opacity-50 flex justify-center items-center   ${
```
Comment (tomershvueli) [inline src/components/Modal/Modal.tsx]: nit: can remove this whitespace change.

**Comment by tomershvueli** [comment]
This is a great start!! 

Few things: 
1. Can we make the modal scrollable vertically when the content overflows? See screenshot where user has a simulated short viewport: 
image
2. Be sure to sync this with `main`, I merged in your Form components, so that will help reduce the size of this PR since those components exist in the repo now.

**Comment by raymondafan** [comment]
Got it just finished the scrollable view @tomershvueli !

**Review by tomershvueli — APPROVED** [review]
This looks fantastic! Great work @raymondafan ! There's a minor merge conflict, once that's resolved we can go ahead and merge this in.
